### PR TITLE
Mozillians multiple emails

### PR DESCRIPTION
**Pull request by akatsoulas**

### CODE REVIEW

**Comment by akatsoulas** [comment]
@johngian one more for you :)

**Comment by akatsoulas** [comment]
Fixes #27, #54, #179.

File: feedthefox/users/adapters.py [hunk 1]
```diff
@@ -10,6 +11,68 @@
...
+
+        # Add alternate emails
+        mozillian_emails = []
+        mozillian_emails.append(mozillian_profile['email'].get('value'))
+        for email_addr in mozillian_profile['alternate_emails']:
+            mozillian_emails.append(email_addr['email'])
+
+        # Lets remove the emails that we already have
+        social_email_addresses = [addr.email for addr in sociallogin.email_addresses]
+        missing_emails = list(set(mozillian_emails) ^ set(social_email_addresses))
```
Comment (johngian) [inline feedthefox/users/adapters.py]: You don't need to convert this to list. You can just iterate over the `set`.

File: feedthefox/users/adapters.py [hunk 2]
```diff
@@ -10,6 +11,68 @@
...
+
+
+def get_mozillian_profile(emails):
+    """Return a profile from mozillians.org give a list of emails."""
+
+    # Check that settings for mozillians.org API are present
+    if not settings.MOZILLIANS_API_URL or not settings.MOZILLIANS_API_KEY:
+        raise ImproperlyConfigured('No API_KEY or URL configured. '
+                                   'Please provide MOZILLIANS_API_URL and '
+                                   'MOZILLIANS_API_KEY in the .env file')
```
Comment (johngian) [inline feedthefox/users/adapters.py]: Please ommit `in the .env file`. Configuration can be anything that changes the host ENV vars.

**Comment by johngian** [comment]
r+wc

File: feedthefox/users/adapters.py [hunk 3]
```diff
@@ -92,15 +123,31 @@ def pre_social_login(self, request, sociallogin):
...
+            # If this is a new social account, connect it please
             sociallogin.user = user
+
+            # Add additional emails, if any
+            addresses, primary = cleanup_email_addresses(request, sociallogin.email_addresses)
+
+            for address in addresses:
+                address.user = user
+                address.save()
+            if not primary:
```
Comment (johngian) [inline feedthefox/users/adapters.py:156]: Please add an empty line here before `if`.

File: feedthefox/users/adapters.py [hunk 4]
```diff
@@ -92,15 +123,31 @@ def pre_social_login(self, request, sociallogin):
...
+                primary_emails = [addr for addr in sociallogin.email_addresses
+                                  if addr.primary]
+                if primary_emails:
```
Comment (johngian) [inline feedthefox/users/adapters.py:159]: Same here.

File: feedthefox/users/adapters.py [hunk 5]
```diff
@@ -92,15 +123,31 @@ def pre_social_login(self, request, sociallogin):
...
+                    # The last is primary
```
Comment (johngian) [inline feedthefox/users/adapters.py]: Both are primaries. Please change the comment to `The last is the primary email from mozillians.org`.

File: feedthefox/users/adapters.py [hunk 6]
```diff
@@ -92,15 +123,31 @@ def pre_social_login(self, request, sociallogin):
...
+                    primary = primary_emails[-1]
+
+            if primary:
```
Comment (johngian) [inline feedthefox/users/adapters.py]: Please remove `if primary`. We always have a primary email.

**Comment by johngian** [comment]
Please fix missing empty space in commit message comment.

**Comment by johngian** [comment]
Tested locally:
- Login with non existing in mozillians.org persona account
  - New inactive account created as expected
- Login with persona using an existing mozillians.org account
  - New active account created, populated with all primary+alternate mozillians.org emails

**Comment by johngian** [comment]
When I tried loggin in with an email already existing in DB but newly added in a mozillians.org profile I am getting a 500 error.

**Comment by johngian** [comment]
r+wc Please check locally and merge.

**Comment by johngian** [comment]
r+

**Pull request merged**